scripts/pnpm-migration/pnpm-workspace-setup-install.mjs

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env node
+import { execSync } from 'child_process';
+import { existsSync, mkdirSync, promises as fs } from 'fs';
+import os from 'os';
+import path from 'path';
+
+const version = '10.11.1';
+const targetDir = path.resolve('./target/pnpm');
+const pnpmPath = path.join(targetDir, 'pnpm');
+const INSTALL_SCRIPT = path.resolve('./scripts/pnpm-migration/pnpm-install-dependencies.mjs');
+const REFRESH_FLAG = path.resolve('.yarn-refreshed');
+
+// ✅ Check if Yarn refresh already happened
+function hasYarnBeenRefreshed() {
+  return existsSync(REFRESH_FLAG);
+}
+
+// 📝 Mark refresh done
+async function markYarnRefreshed() {
+  await fs.writeFile(REFRESH_FLAG, 'ok\n');
+}
+
+// 🧹 Cleanup flag on second run
+async function clearYarnRefreshFlag() {
+  if (existsSync(REFRESH_FLAG)) {
+    await fs.unlink(REFRESH_FLAG).catch(() => {});
+    console.log('🧹 Yarn refresh flag cleared. Ready for next cycle.');
+    process.exit(0);
+  }
+}
+
+// Safe cleanup on interrupt signals
+process.on('SIGINT', () => {
+  console.log('\n🛑 Caught SIGINT. Cleaning up...');
+  clearYarnRefreshFlag();
+});
+process.on('SIGTERM', () => {
+  console.log('\n🛑 Caught SIGTERM. Cleaning up...');
+  clearYarnRefreshFlag();
+});
+
+// ⬇️ Download and install PNPM binary directly into targetDir
+function installPnpm() {
+  console.log(`⬇️ Downloading PNPM v${version} binary...`);
+  mkdirSync(targetDir, { recursive: true });
+
+  const platform = os.platform();
+  const arch = os.arch();
+
+  let binaryName = '';
+  if (platform === 'darwin') {
+    binaryName = arch === 'arm64' ? 'pnpm-macos-arm64' : 'pnpm-macos-x64';
+  } else if (platform === 'linux') {
+    binaryName = arch === 'arm64' ? 'pnpm-linux-arm64' : 'pnpm-linux-x64';
+  } else if (platform === 'win32') {
+    binaryName = 'pnpm-win.exe';
+  } else {
+    console.error(`❌ Unsupported platform: ${platform}-${arch}`);
+    process.exit(1);
+  }
+
+  const url = `https://github.com/pnpm/pnpm/releases/download/v${version}/${binaryName}`;
+  const outputPath = pnpmPath + (platform === 'win32' ? '.exe' : '');
+
+  try {
+    execSync(`curl -fsSL ${url} -o ${outputPath}`, { stdio: 'inherit' });
+    execSync(`chmod +x ${outputPath}`);
```

Comment:
is this script meant to be started only on CI or locally as well ?

i believe chmod won't be portable on win32 platform